scio-test/src/main/scala/com/spotify/scio/testing/PipelineTestUtils.scala

```diff
@@ -124,14 +124,23 @@ trait PipelineTestUtils {
   : Seq[U] = {
     runWithLocalOutput { sc =>
       fn(sc.parallelize(data1), sc.parallelize(data2), sc.parallelize(data3), sc.parallelize(data4))
-    }
+    }._2
   }
 
-  private def runWithLocalOutput[U](fn: ScioContext => SCollection[U]): Seq[U] = {
+  /**
+   * Test pipeline components with materialized resulting collection.
+   *
+   * The result [[com.spotify.scio.values.SCollection SCollection]] from `fn` is extracted and to be
+   * verified.
+   *
+   * @param fn transform to be tested
+   * @return Scio Result Object and Job results in an in-memory Scala list
```

Comment:
Is something like this more clear? `@return a tuple containing the [[ScioResult]] and the materialized result of fn as a [[Seq[T]]]`